core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -132,16 +133,16 @@ public static ViewMetadata fromJson(JsonNode json) {
       historyEntries.add(ViewHistoryEntryParser.fromJson(vLog));
     }
 
-    return ImmutableViewMetadata.builder()
-        .location(location)
-        .currentVersionId(currentVersionId)
-        .properties(properties)
-        .versions(versions)
-        .schemas(schemas)
-        .currentSchemaId(currentSchemaId)
-        .history(historyEntries)
-        .formatVersion(formatVersion)
-        .build();
+    return ImmutableViewMetadata.of(
+        formatVersion,
+        location,
+        currentSchemaId,
+        schemas,
+        currentVersionId,
+        versions,
+        historyEntries,
+        properties,
+        ImmutableList.of());
```

Comment:
What about the view metadata location? We track that in `TableMetadata` so we can determine if it changed.

Reply:
we haven't been tracking that for Views so far but it was on my list to be added once we need it. That being said, I'll handle this in a separate PR